src/plugins/links/public/embeddable/initialize_links.ts

```diff
@@ -0,0 +1,129 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fastIsEqual from 'fast-deep-equal';
+import { BehaviorSubject, distinctUntilKeyChanged, filter, switchMap } from 'rxjs';
+import {
+  apiPublishesPanelDescription,
+  apiPublishesPanelTitle,
+  apiPublishesSavedObjectId,
+  StateComparators,
+} from '@kbn/presentation-publishing';
+import { apiIsPresentationContainer } from '@kbn/presentation-containers';
+import { PanelIncompatibleError, PanelNotFoundError } from '@kbn/embeddable-plugin/public';
+import { LinksAttributes } from '../../common/content_management';
+import { LinksParentApi, LinksSerializedState, ResolvedLink } from './types';
+import { resolveLinks } from './utils';
+import { openEditorFlyout } from '../editor/open_editor_flyout';
+import { loadFromLibrary } from '../content_management';
+import { CONTENT_ID } from '../../common';
+
+const isParentApiCompatible = (parentApi: unknown): parentApi is LinksParentApi =>
+  apiIsPresentationContainer(parentApi) &&
+  apiPublishesSavedObjectId(parentApi) &&
+  apiPublishesPanelTitle(parentApi) &&
+  apiPublishesPanelDescription(parentApi);
+
+export async function initializeLinks(
+  state: LinksSerializedState,
+  uuid: string,
+  parentApi: unknown
+) {
+  if (!isParentApiCompatible(parentApi)) throw new PanelIncompatibleError();
+
+  const error$ = new BehaviorSubject<Error | undefined>(undefined);
+  const isEditable$ = new BehaviorSubject<boolean>(true);
+
+  let attributes: LinksAttributes | undefined;
+  try {
+    const loadedState = state.savedObjectId ? await loadFromLibrary(state.savedObjectId) : state;
+    if (loadedState.attributes) attributes = loadedState.attributes;
+  } catch (e) {
+    error$.next(new PanelNotFoundError());
+    isEditable$.next(false);
+  }
+
+  const resolvedLinks$ = new BehaviorSubject<ResolvedLink[]>([]);
+
+  const attributes$ = new BehaviorSubject<LinksAttributes | undefined>(attributes);
+  const savedObjectId$ = new BehaviorSubject<string | undefined>(state.savedObjectId);
+
+  const defaultPanelTitle = new BehaviorSubject<string | undefined>(attributes?.title);
+  const defaultPanelDescription = new BehaviorSubject<string | undefined>(attributes?.description);
+
+  attributes$
+    .pipe(
+      filter(Boolean),
+      distinctUntilKeyChanged('links'),
+      switchMap(({ links }) => resolveLinks(links))
+    )
+    .subscribe({
+      next: (resolvedLinks) => resolvedLinks$.next(resolvedLinks),
+      error: (error) => error$.next(error),
+    });
+
+  const onEdit = async () => {
+    try {
+      const newState = await openEditorFlyout({
+        initialState: state,
+        parentDashboard: parentApi,
+      });
+      parentApi
+        .replacePanel(uuid, {
+          panelType: CONTENT_ID,
+          initialState: newState,
+        })
+        .catch((e: Error) => {
+          error$.next(e);
+        });
```

Comment:
It would be nice if we didn't have to replace the panel every time and we could just update the `attributes` directly 🤔 that's how we do it for the image embeddable, but it might be more complicated here because the links panel can be by-reference

Reply:
With the latest updates we now populate the editor with LinksRuntimeState. Even by reference panels are updated in the background as we use runtime state for the UI.